knowledge_engine/knowledge_engine/stat/runner.py

```
@@ -0,0 +1,574 @@
+        error_msg = ", ".join(failed_names) if failed_names else None
+    except SoftTimeLimitExceeded as exc:
+        # Task-level soft timeout: persist the run as failed, then re-raise so
+        # Celery sees the failure (retry/alert machinery, acks_late requeue).
+        # The generic Exception branch below must not swallow this — without
+        # the re-raise, collect_all returns normally and Celery marks a
+        # timed-out task as SUCCESS, defeating the soft_time_limit contract.
+        abort_exc = exc
+        run_status = "failed"
+        error_msg = "soft time limit exceeded"
+        logger.error(f"[kb_stat] run_id={run_id} soft time limit exceeded")
+    except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)